### DISCUSSION TITLE
HealthChecks integration?

### DISCUSSION THREAD

**Discussion by fleed** [ThreeMammals/Ocelot#1901] [Ideas]
Are there any "utility" methods to easily integrate health checks?

I would like to expose an "aggregated" health status that considers all the health status information coming from the services behind the gateway.

I would expect to configure the health endpoints in the ocelot configuration together with the routes.

**Comment by jducobu** [comment]
Is this feature already exist?

**Reply by raman-m** [reply]
No!

**Comment by raman-m** [comment]
@fleed
 Hi Francesco!
**Will you contribute?**

> Are there any "utility" methods to easily integrate health checks?

No, there are not!
Why not just to define Health Check route as a pair for a route? You can do always.
But duplication is bad, and it would be better to introduce special route property. And internal background logic can monitor health of the route, not waiting for a real request.
